frontend/pages/occurrences/[slug].tsx

```diff
@@ -0,0 +1,51 @@
+import { OccurrenceModule } from "@/interfaces";
+import { GetStaticPaths, GetStaticProps } from "next";
+import { FC } from "react";
+
+import Layout from "@/components/Layout";
+import ModuleContainer from "@/components/moduledetails/ModuleContainer";
+import axios from "axios";
+
+interface OccurrenceProps {
+  occurrence: OccurrenceModule;
+}
+
+/**
+ * A page that renders the HTML of a single occurrence.
+ */
+const Occurrence: FC<OccurrenceProps> = ({ occurrence }: OccurrenceProps) => {
+  return (
+    <Layout title={""}>
+      <ModuleContainer module={occurrence} />
+    </Layout>
+  );
+};
+export default Occurrence;
+
+export const getStaticProps: GetStaticProps = async ({ params }) => {
+  let occurrence = {};
+  const { slug } = params;
+
+  await axios
+    .get(`http://django:8000/api/occurrences/${slug}/`)
+    .then((response) => {
+      occurrence = response.data;
+    })
+    .catch((error) => {
+      // TODO: how should we handle errors here?
```

Comment:
What kind of errors do we expect here?

I guess it will return some kind of 'not found' error in case of invalid/nonexistent slugs. For 'not found' errors, we probably need to redirect to 404 pages on the client. We could generalize this for other endpoints with middleware maybe?

Reply:
Yeah I created [`MH-196`](https://modularhistory.atlassian.net/browse/MH-196) recently for creating [custom error pages](https://nextjs.org/docs/advanced-features/custom-error-page). I don't know what the best way to redirect to those pages is, though. My first thought would be to just pass the error along to the page component add a `useEffect` hook to redirect to the `404`/`500` page if an error exists, but this might not be the most efficient method.